### ISSUE TITLE
Audit headline font sizes

### ISSUE THREAD

**Issue by robertdenton** [registerguard/bulldog#209]
Need to look at current font-sizes and how they fit to baseline. I'm afraid I might have busted that at some point since Micky left. This is part of the greater homepage look audit.

## TODO

* [x] Remove media query for h1/h2 font sizes (type.less 36-47)
* [x] Add missing equations to type.less h1 and h5
* [ ] Drop lead headline from h3 to h4 maybe? This is not high priority.

**Comment by robertdenton** [comment]
Font size goods are in two places. There are some base variables set in var.less but most of it comes from type.less.

## var.less

Our base font size is [1em](https://github.com/registerguard/bulldog/blob/master/build/files/css/less/var.less#L83) - [which on most browsers is 16px](https://developer.mozilla.org/en-US/docs/Web/CSS/font-size#Pixels). 

At first I wasn't sure why we wouldn't set it specifically to 16px to try and control different defaults for different browsers, but I think doing so might negatively impact site accessibility for people/browsers who have trouble reading and manually increase the browser's default font size. We'll leave it as it is.

The base line height is [1.5em (24px)](https://github.com/registerguard/bulldog/blob/master/build/files/css/less/var.less#L85).

## type.less

This is where it gets tricky. For the purposes of this issue, we're just going to look at headline and deck styles, but there are other font-size concerns in this file.

We open by normalizing all headline and subhed styles to the baseFontSize variable (1em/16px).

From there it goes from h1-h6 and decreases font-size and line-height based on a common equation (more on this later).

Below that, there is a media query to increase the h1 and h2 sizes on large desktops (greater than 1345px). This seems like bad practice to me, and I was the one that added it back in March 2015 upon request from CD. I would like to remove that as part of this ticket.

### Back in time

In order to figure out how I might have messed up the baseline, I'm going back through commits that touched var.less and type.less. 

var.less has only had [four commits](https://github.com/registerguard/bulldog/commits/master/build/files/css/less/var.less) and none of them were made by me. That's good, that means I didn't touch the base font size variable.

type.less has [21 commits (11 by me)](https://github.com/registerguard/bulldog/commits/master/build/files/css/less/type.less).

* 1/27/15
  * Starting from my first commit, we see a few commits that do not affect headline sizes. 
* 2/25/15
  * Adding tools widgets, no headline styles
* 3/6/15
  * **Add media query for h1-h6 headline font sizes on screens larger than 1345px**
* 3/9/15
  * **Roll back h3-h6 sytles, keep h1, h2**
* 4/14/15
  * Blockquote styles tweaks
* 5/17/16
  * Red hed hover darken
* 3/3/17
  * Add sponsor widget styles (not sure why in this file but we'll let it ride...)

So, good news, I didn't mess it up nearly as bad as I feared.

![whew](https://media.tenor.com/images/e84ece8fa9ba81f9443e1e7de112e3a8/tenor.gif)

As long as I roll back the media query and don't mess with the equations we should be doing alright.

### While I'm here ...

While I'm here, I've always wanted to know how Micky came up with these funky line heights, I may as well try to figure it out real quick.

I did some quick 2012 [line height reading](https://www.google.com/search?q=line+height+equation+%2B+baseline+grid+%2B+3+%2F&ie=utf-8&oe=utf-8&client=firefox-b-1-ab):

* [Baseline Grid Workflow (How To) - 11/28/12](https://xavierbertels.com/blog/baseline-grid-workflow) (kind of the opposite of Micky's equation)
* [Baseline grid calculator](http://topfunky.com/baseline-rhythm-calculator/) (doesn't give equation)
* [CSS Baseline: The Good, The Bad And The Ugly - 12/17/12](https://www.smashingmagazine.com/2012/12/css-baseline-the-good-the-bad-and-the-ugly/) (skimmed)
* [4 simple steps to vertical rhythm - 2/21/12](http://typecast.com/blog/4-simple-steps-to-vertical-rhythm) (a little simplistic)

Not a whole lot of help, but gives me some context. Going to try and find patterns (not including the media queries that I want to remove).

Table of findings, see spreadsheet [here](https://docs.google.com/spreadsheets/d/1TCPeGge2Q6Dz5zCvGqzsjVLvbwXpajl5mL-0dpAhUTc/edit?usp=sharing).

| hed | em | px | line-height | equation |
| --- | --- | --- | --- | --- |
| h1|2.875|46|1.043478261|(((3 / 2.875) = 1.043478261) * 1) |
| h2|2.5|40|1.05|(((3 / 2.5) = 1.2) * .875) |
| h3|2.125|34|1.058823529|(((3 / 2.125em) = 1.41176470588235) * .75) |
| h4|1.75|28|1.071428571|(((3 / 1.75em) = 1.71428571428571) * .625) |
| h5|1.375|22|1.090909091|(((3 / 1.375em) = 2.181818182) * .5) |
| h6|1.125|18|1.166666667|(((1.5 / 1.125) = 1.33333333333333) * .875) |

Long story short, h1-h5 have a 6px difference and drop equation coefficient .125 (1-.5) to find line height.

On h6, it only drops 4px (I'm guessing to keep it greater than 16px), and bumps the coefficient to .875 (probably to continue the airing out trend of the line-height).

It kind of bugs me that h6 doesn't just drop down to 16px. It could be the same size, but styled differently. This might be something to think about.

**Comment by robertdenton** [comment]
Now, the next question is: Can we tighten up these headlines by only jumping 4px instead of 6px? What was the thought process behind 6 and does that screw up the baseline grid? It seems to me that 4px jumps would fit the grid better because that's .25 of 1em (16px) instead of .375 of 1em.

Here's what that math might look like:


| hed | em | px | line-height | equation |
| --- | --- | --- | --- | --- |
| h1 | 2.25 | 36 | 0.8888888889 | (((2 / 2.25) = 1.333333333) * 1) | 
| h2 | 2 | 32 | 0.875 | (((2 / 2) = 1.5) * .875) | 
| h3 | 1.75 | 28 | 0.8571428571 | (((2 / 1.75em) = 1.714285714) * .75) | 
| h4 | 1.5 | 24 | 0.8333333333 | (((2 / 1.5em) = 2) * .625) | 
| h5 | 1.25 | 20 | 0.8 | (((2 / 1.25em) = 2.4) * .5) | 
| h6 | 1 | 16 | 0.75 | (((2/ 1) = 3) * .875) | 

Numbers sure are a lot cleaner ... Let's see how it looks ... First image has 6px jumps, the second has 4px.

![6px](https://user-images.githubusercontent.com/4853944/33735935-b4859314-db45-11e7-9080-1345cac724e0.jpg)
![4px](https://user-images.githubusercontent.com/4853944/33735936-b4a13592-db45-11e7-9b3e-f10baa0f2c29.jpg)

It's definitely tighter horizontally, but I'm not sure the large heads have enough weight for story pages. I do like having the smaller 16px headline size. ~~Another thing is that the vertical space is a bit much. If we were to use this, we should use a base of 2 to divide.~~ This was updated on the second go.
![8](https://user-images.githubusercontent.com/4853944/33736922-4e2e4a08-db49-11e7-8d24-836cf4cd9167.jpg)
![6](https://user-images.githubusercontent.com/4853944/33736923-4e75d864-db49-11e7-9a08-87b7f67a6872.jpg)
![4](https://user-images.githubusercontent.com/4853944/33736924-4e979a30-db49-11e7-872b-3a268675782b.jpg)




Let's try 8px jumps starting at 16 with a base of 4 to divide.

| hed | em | px | line-height | equation |
| --- | --- | --- | --- | --- |
| h1 | 3.5 | 56 | 1.142857143 | (((4 / 2.25) = 1.333333333) * 1) |
| h2 | 3 | 48 | 1.166666667 | (((4 / 2) = 1.5) * .875) |
| h3 | 2.5 | 40 | 1.2 | (((4 / 1.75em) = 1.714285714) * .75) |
| h4 | 2 | 32 | 1.25 | (((4 / 1.5em) = 2) * .625) |
| h5 | 1.5 | 24 | 1.333333333 | (((4 / 1.25em) = 2.4) * .5) |
| h6 | 1 | 16 | 1.5 | (((4/ 1) = 3) * .875) |

**Comment by robertdenton** [comment]
For the sake of complete documentation, this is the size difference between the h1/h2 media queries and how they should look. There's not much of a vertical difference, but it is much wider.

If we want to do media query size changes, we should bump everything 2px or 4px or something that makes sense.

![mediaquery](https://user-images.githubusercontent.com/4853944/33782413-81f15e1a-dc0d-11e7-8f02-87db15b3e8d0.jpg)
